x-pack/plugins/alerting/server/usage/lib/get_telemetry_from_event_log.ts

```diff
@@ -0,0 +1,620 @@
+              },
+            },
+            aggs: {
+              by_reason: {
+                terms: {
+                  field: 'event.reason',
+                  size: NUM_ALERTING_EXECUTION_FAILURE_REASON_TYPES,
+                },
+              },
+            },
+          },
+          by_rule_type_id: {
+            terms: {
+              field: 'rule.category',
+              size: NUM_ALERTING_RULE_TYPES,
+            },
+            aggs: {
+              ...executionTimeAggs,
+              execution_failures: {
+                filter: {
+                  term: {
+                    'event.outcome': 'failure',
+                  },
+                },
+                aggs: {
+                  by_reason: {
+                    terms: {
+                      field: 'event.reason',
+                      size: NUM_ALERTING_EXECUTION_FAILURE_REASON_TYPES,
+                    },
+                    aggs: {
+                      by_rule_type_id: {
+                        terms: {
+                          field: 'rule.category',
+                          size: NUM_ALERTING_RULE_TYPES,
+                        },
+                      },
+                    },
+                  },
+                },
+              },
+              percentile_scheduled_actions: generatedActionsPercentilesAgg,
+              percentile_alerts: alertsPercentilesAgg,
+            },
+          },
+        },
+      },
+    };
+
+    logger.debug(`query for getExecutionsPerDayCount - ${JSON.stringify(query)}`);
+    const results = await esClient.search(query);
+
+    logger.debug(`results for getExecutionsPerDayCount query - ${JSON.stringify(results)}`);
+
+    const totalRuleExecutions =
+      typeof results.hits.total === 'number' ? results.hits.total : results.hits.total?.value;
+
+    const aggregations = results.aggregations as {
+      by_rule_type_id: AggregationsTermsAggregateBase<GetExecutionCountsAggregationBucket>;
+      execution_failures: IGetExecutionFailures;
+      percentile_scheduled_actions: AggregationsPercentilesAggregateBase;
+      percentile_alerts: AggregationsPercentilesAggregateBase;
+      avg_execution_time: AggregationsSingleMetricAggregateBase;
+      avg_es_search_duration: AggregationsSingleMetricAggregateBase;
+      avg_total_search_duration: AggregationsSingleMetricAggregateBase;
+    };
+
+    const aggregationsByRuleTypeId: AggregationsBuckets<GetExecutionCountsAggregationBucket> =
+      aggregations.by_rule_type_id.buckets as GetExecutionCountsAggregationBucket[];
+
+    return {
+      ...parseRuleTypeBucket(aggregationsByRuleTypeId),
+      ...parseExecutionFailureByRuleType(aggregationsByRuleTypeId),
+      ...parseExecutionCountAggregationResults(aggregations),
+      countTotalRuleExecutions: totalRuleExecutions ?? 0,
+    };
+  } catch (err) {
+    logger.warn(
+      `Error executing alerting telemetry task: getExecutionsPerDayCount - ${JSON.stringify(err)}`
```

Comment:
I don't think this will print the stack, but we'll want to see it if an error is thrown, in case one of the casts isn't quite right ...

Reply:
Added stack trace to `meta` in this commit: https://github.com/elastic/kibana/pull/134769/commits/fb0de8634a64a5872fdbd3f0f9d819e189ffd83b